### PR TITLE
SPARK-9059 Update Direct Kafka Word count examples to show the use of…

### PR DESCRIPTION
**Pull request by kico445**
SPARK-9059 Update Direct Kafka Word count examples to show the use of HasOffsetRanges

### CODE REVIEW

**Comment by koeninger** [comment]
I thought the idea for this ticket was to have separate examples for accessing offset ranges.

The complexity of offsets doesn't really have anything to do with a basic word count.

**Comment by BenFradet** [comment]
Also, I think it was meant to present an example in python.

File: examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java [hunk 1]
```diff
@@ -106,6 +111,32 @@ public Integer call(Integer i1, Integer i2) {
...
+    messages.transformToPair(
+        new Function<JavaPairRDD<String, String>, JavaPairRDD<String, String>>() {
+            @Override
+            public JavaPairRDD<String, String> call(JavaPairRDD<String, String> rdd) throws Exception {
+            OffsetRange[] offsets = ((HasOffsetRanges) rdd.rdd()).offsetRanges();
+            offsetRanges.set(offsets);
+            return rdd;
+            }
+        }
+    ).foreachRDD(
```
Comment (jerryshao) [inline examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java]: I think we could get and print out the offsetRanges immediately in `foreachRDD`, not sure why we need another `transformToPair` to only get the offsetRanges?

**Comment by srowen** [comment]
I believe this PR/JIRA should be closed as a duplicate

**Comment by BenFradet** [comment]
@srowen There is still the issue of the python example though.

**Comment by srowen** [comment]
Is that what the JIRA is about? OK if we want to make it so, otherwise it's a duplicate. Change the text though.

This PR however is not about Python and is not open vs master. Do you mind closing this PR?

File: examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java (hunk 1 above)
Comment (tdas) [inline examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java]: @kico445 I agree with @jerryshao Could you please update the PR?

File: examples/src/main/scala/org/apache/spark/examples/streaming/DirectKafkaWordCount.scala [hunk 2]
```diff
@@ -65,6 +65,18 @@ object DirectKafkaWordCount {
     val wordCounts = words.map(x => (x, 1L)).reduceByKey(_ + _)
     wordCounts.print()
 
+    //Access the offset ranges using HasOffsetRanges
+    // Hold a reference to the current offset ranges, so it can be used downstream
+    var offsetRanges = Array[OffsetRange]()
+    messages.transform { rdd =>
```
Comment (tdas) [inline examples/src/main/scala/org/apache/spark/examples/streaming/DirectKafkaWordCount.scala]: Same comment as above for

**Comment by srowen** [comment]
Agree with the comments from @tdas but this also needs to be reopened vs `master` anyway. Can you close this PR and make a new one?

**Comment by tdas** [comment]
@kico445 If you are not able to address the comments, mind closing this PR?

**Pull request closed without merging**